packages/toolpad-components/src/Autocomplete.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import {
+  Autocomplete as MuiAutocomplete,
+  AutocompleteProps as MuiAutocompleteProps,
+  TextField,
+} from '@mui/material';
+import { createComponent } from '@mui/toolpad-core';
+
+type AutocompleteOption = string | { label?: string };
+
+interface AutocompleteProps
+  extends Omit<MuiAutocompleteProps<AutocompleteOption, false, false, false>, 'renderInput'> {
+  options: AutocompleteOption[];
+  label: string;
+}
+
+function Autocomplete({ options, label, ...rest }: AutocompleteProps) {
+  return (
+    <MuiAutocomplete
+      options={options ?? []}
+      renderInput={(params) => <TextField {...params} label={label} variant="outlined" />}
+      {...rest}
+    />
+  );
+}
+
+export default createComponent(Autocomplete, {
+  layoutDirection: 'both',
+  argTypes: {
+    options: {
+      typeDef: { type: 'array' },
```

Comment:
Should this be the same as the select component?

Reply:
I tried both ways and realised that this works better, since the `options` prop for the Autocomplete component is more flexible about what it can take as an input. A series of strings, or an array of objects with a `label` prop. In this scenario, a simple JSON Editor works better in my opinion

Reply:
> A series of strings, or an array of objects with a `label` prop.

Isn't that the same as the select component?. Seems to work when I bind to it.

Reply:
The Select component treats the string as `values`, not `label`s

Reply:
I can bind the following to a Select and it recognizes all of these as a valid option:
```ts
[{ value: "foo", label: "Foo" }, "bar", { value: "baz" }]
```

Reply:
I was referring to this issue when using the UI of the Select options editor:

https://user-images.githubusercontent.com/19550456/207851957-f24b8b06-6891-4148-abdc-4439f70dda51.mov

The `value` prop should be the `value` attribute of the option, not the `label`